### PR TITLE
Map task list tab to URL

### LINKED ISSUE #878: Projects - Add Routing to Task List
### Description

**If #448 is still open then branch from/PR into `feature/#448-task-lists` feature branch, otherwise work with `develop` as usual.**

Look into getting the tabs in the Task List to route to URL. For example, maybe if we navigate to the DONE tab of the task list we can have the URL go to `/projects/1.1.0/2` or  `/projects/1.1.0?taskView=2` or something like that. 

From James ([source](https://github.com/Northeastern-Electric-Racing/FinishLine/pull/830#pullrequestreview-1294623125)):

> ...I definitely think we should have [task list routing]. Generally speaking I'm a fan of embedding app state in the URL to ensure deep linking is possible

### Acceptance Criteria

If I click any tab in the task list and then copy the current URL, I should be able to open the URL somewhere else and have the same tab open. (Tab state is stored in the URL)

### Proposed Solution

Use routing to bind the active tab to the URL. You'll need to do some research on how react router and [MUI tab routing](https://mui.com/material-ui/guides/routing/#tabs) work. Make good design decisions (code simplicity, user friendliness, etc) on that format you want the URL to look like.

### Mocks

IMG_7651

### PR DESCRIPTION
**Pull request by ninest**
## Changes

Map MUI task list tab to URL in the projects page.

## Screenshots

Default to in-progress tab: 
image

Clicking on tabs: 
image

image

Reloading:
image

## To Do

- [x] Allow user to switch between tabs using browser forward/back buttons

## Checklist

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #878

### CODE REVIEW

**Comment by ninest** [comment]
If the user presses the browser's back button, the URL changes but the tab doesn't. This is fixed with this `useEffect`:

```tsx
const tabUrlValues = ['in-backlog', 'in-progress', 'done'];
const match = useRouteMatch<{ wbsNum: string; tabValueString: string }>('/projects/:wbsNum/:tabValueString');
const tabValueString = match?.params?.tabValueString;
...

const { pathname } = useLocation();
useEffect(() => {
  const newTabValue: number = tabUrlValues.indexOf(tabValueString ?? 'in-progress');
  setValue(newTabValue);
}, [pathname]);
```

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx [hunk 1]
```diff
@@ -25,7 +27,25 @@ const TaskList = ({ tasks, currentWbsNumber, defaultClosed, team, hasTaskPermiss
   const auth: Auth = useAuth();
   const taskListTitle: string = 'Task List';
 
-  const [value, setValue] = useState<number>(1);
+  // Values that go in the URL depending on the tab value, example /projects/0.0.0/in-progress
+  const tabUrlValues = ['in-backlog', 'in-progress', 'done'];
+
+  const match = useRouteMatch<{ wbsNum: string; tabValueString: string }>('/projects/:wbsNum/:tabValueString');
```
Comment (Ari-S-123) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx:33]: `${routes.PROJECTS}`

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx [hunk 2]
```diff
@@ -25,7 +27,25 @@ const TaskList = ({ tasks, currentWbsNumber, defaultClosed, team, hasTaskPermiss
...
+  // Default to the "in-progress" tab
+  const initialValue: number = tabUrlValues.indexOf(tabValueString ?? 'in-progress');
+  const [value, setValue] = useState<number>(initialValue);
+
+  // Change tab when the browser forward/back button is pressed
+  const { pathname } = useLocation();
+  useEffect(() => {
+    const newTabValue: number = tabUrlValues.indexOf(tabValueString ?? 'in-progress');
+    setValue(newTabValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Ari-S-123) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx:46]: We should not be disabling eslint anywhere.

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]

**Review by anthonybernardi — APPROVED** [review]
awesome!!

**Pull request merged**